pkg/acquisition/modules/file/file.go

```diff
@@ -0,0 +1,386 @@
+package fileacquisition
+
+import (
+	"bufio"
+	"compress/gzip"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/crowdsecurity/crowdsec/pkg/acquisition/configuration"
+	leaky "github.com/crowdsecurity/crowdsec/pkg/leakybucket"
+	"github.com/crowdsecurity/crowdsec/pkg/types"
+	"github.com/fsnotify/fsnotify"
+	"github.com/nxadm/tail"
+	"github.com/pkg/errors"
+	"github.com/prometheus/client_golang/prometheus"
+	log "github.com/sirupsen/logrus"
+	"golang.org/x/sys/unix"
+	"gopkg.in/tomb.v2"
+	"gopkg.in/yaml.v2"
+)
+
+var linesRead = prometheus.NewCounterVec(
+	prometheus.CounterOpts{
+		Name: "cs_filesource_hits_total",
+		Help: "Total lines where read.",
+	},
+	[]string{"source"})
+
+type FileConfiguration struct {
+	Filenames                         []string
+	Filename                          string
+	ForceInotify                      bool `yaml:"force_inotify"`
+	configuration.DataSourceCommonCfg `yaml:",inline"`
+}
+
+type FileSource struct {
+	config             FileConfiguration
+	watcher            *fsnotify.Watcher
+	watchedDirectories map[string]bool
+	tails              map[string]bool
+	logger             *log.Entry
+	files              []string
+}
+
+func (f *FileSource) Configure(Config []byte, logger *log.Entry) error {
+	fileConfig := FileConfiguration{}
+	f.logger = logger
```

Comment:
I think we want to customize logger here (with ie. pattern or unique identifier to be able to track)